File: src/Frontend/src/types/item.ts
export interface Item {
  id: string;
  name: string;
  description?: string;
  quantity: number;
  price: number;
  createdAt: string;
}

export type CreateItemInput = Omit<Item, 'id' | 'createdAt'>;
export type UpdateItemInput = Partial<CreateItemInput>;

File: src/Backend/Models/Item.cs
namespace InventorySystem.Models;

public class Item
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public int Quantity { get; set; }
    public decimal Price { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}